lib/cog/pipeline/signals.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Cog.Pipeline.DoneSignal do
+  defstruct [:error, :invocation, :bundle_version_id, :template]
+
+  def error?(%__MODULE__{error: nil}), do: false
+  def error?(%__MODULE__{}), do: true
+
+  def done?(%__MODULE__{}), do: true
+  def done?(_), do: false
```

Comment:
This is a catch-all for "other signals", right?

If so, does `error?/1` need to have a similar catch-all function head?